src/backend/halo2.rs

```diff
@@ -305,3 +310,110 @@ pub fn to_halo2_advice<F: Field>(
         _ => panic!("jarll wrong phase"),
     }
 }
+
+#[derive(Clone, Default)]
+pub struct ChiquitoHalo2Circuit<F: Field + From<u64>> {
+    compiled: ChiquitoHalo2<F>,
+    witness: Option<Assignments<F>>,
+}
+
+impl<F: Field + From<u64>> ChiquitoHalo2Circuit<F> {
+    pub fn new(compiled: ChiquitoHalo2<F>, witness: Option<Assignments<F>>) -> Self {
+        Self { compiled, witness }
+    }
+}
+
+impl<F: Field + From<u64> + Hash> h2Circuit<F> for ChiquitoHalo2Circuit<F> {
+    type Config = ChiquitoHalo2<F>;
+
+    type FloorPlanner = SimpleFloorPlanner;
+
+    type Params = ChiquitoHalo2<F>;
+
+    fn without_witnesses(&self) -> Self {
+        Self::default()
+    }
+
+    fn params(&self) -> Self::Params {
+        self.compiled.clone()
+    }
+
+    fn configure_with_params(
+        meta: &mut ConstraintSystem<F>,
+        mut compiled: Self::Params,
+    ) -> Self::Config {
+        compiled.configure(meta);
+
+        compiled
+    }
```

Comment:
We are using configure with params instead of configure. Is it to force the function to take ChiquitoHalo2 objects? A bit confused by Halo2's doc for `configure_wiht_params`:
```
    /// The circuit is given an opportunity to describe the exact gate
    /// arrangement, column arrangement, etc.  Takes a runtime parameter.  The default
    /// implementation calls `configure` ignoring the `_params` argument in order to easily support
    /// circuits that don't use configuration parameters.
```

Reply:
We need to use configure_wiht_params in order to be able to have the same struct represent all chiquito circutis, as normal configure does not accept parameters, we could not give the chiquito circuit to it

Reply:
Got it! Thanks.